File: js/OOP/Inheritance.js
class Animal {
    constructor(species) {
        this.species = species
    }
    makeSound() {
        console.log('making sound')
    }
}

class Dog extends Animal {
    constructor(name) {
        super('Dog',)
        this.name = name
    }
    bark() {
        console.log('bark bark')
    }
}

const d1 = new Dog('Pako')
console.log(d1.name)
d1.makeSound()
d1.bark()

File: python/OOP/Inheritance.py
# parent class
class Animal:
    def __init__(self, species: str, age: int=0):
        self.species = species
        self.age = age
        
    def makeSound(self):
        print('making sound')

# child class
class Dog(Animal):
    def __init__(self, age: int, name: str):
        super().__init__('Dog', age)
        self.name = name
    def bark(self):
        print('bark bark')


d1 = Dog(3, 'pako')

d1.bark()